### PR TITLE
Trip planning initiative front

### PR DESCRIPTION
**Pull request by d3mola**
#### What does this PR do?
This PR Implements the trip planning features.

#### Description of Task to be completed?
- Create travel checklists
- Edit travel checklists
- Delete travel checklists
- Display travel checklists for `Open` requests
- Display travel checklists submission percentage

#### How should this be manually tested?
- A user should be able to view a travel checklist for an `open` request
      - Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
      - Change into the project directory
      - Run the server `make start`
      - Login and make a `new request`
      - Click on the `new request` ellipsis and then click on `travel checklist` on the modal dropdown
      - The travel checklist modal should pop up and show the checklist for the travel request.
- A Travel Admin/Super Admin for a location
    -- should be able to edit a previously set travel checklist and the corresponding output of each 
        checklist item
    -- should be able to delete a checklist item
    -- should be able to create a checklist item

#### Any background context you want to provide?
- You need to clone the [backend](https://github.com/andela/travel_tool_back.git) to test this functionality.

- **Add yourself as travel admin**
   -- Add your email address in the .env file on the backend as `DEFAULT_ADMIN=` and start/restart your server
   -- Grab your token from the browser after login and add the token as authorization on postman
   -- Send a put request to `http://localhost:5000/api/v1/user/admin`

#### What are the relevant pivotal tracker stories?
[#161044514](https://www.pivotaltracker.com/story/show/161044514)
[#161044557](https://www.pivotaltracker.com/story/show/161044557)
[#161044598](https://www.pivotaltracker.com/story/show/161044598)
[#161044635](https://www.pivotaltracker.com/story/show/161044635)
[#161044756](https://www.pivotaltracker.com/story/show/161044756)

#### Screenshots (if appropriate)
screen shot 2018-10-13 at 6 22 22 pm

screen shot 2018-10-13 at 6 23 23 pm

#### Questions:
N/A

### CODE REVIEW

**Comment by samuelkubai** [comment]
@d3mola please resolve conflicts and ask for reviews.

**Comment by samuelkubai** [comment]
@d3mola please resolve the snapshot conflict.

**Comment by Nedson202** [comment]
This has been resolved @samuelkubai

File: src/components/Table/index.jsx [hunk 1]
```diff
@@ -256,7 +294,8 @@ Table.defaultProps = {
   modalType: null,
   message: '',
   page: '',
-  requestId: ''
+  requestId: '',
+  // percentage: ''
```
Comment (seunzone) [inline src/components/Table/index.jsx]: commented out lines of code like this not in use should be removed.

File: src/redux/middleware/commentsSaga.js [hunk 2]
```diff
@@ -27,6 +27,15 @@ export function* createCommentAsync(action) {
   }
 }
 
+export function* watchCreateComment() {
+  yield takeLatest(CREATE_COMMENT, createCommentAsync);
+}
+
+
```
Comment (seunzone) [inline src/redux/middleware/commentsSaga.js]: the multiple lines here should be removed

File: src/views/Checklist/__tests__/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  it('should render the Checklist page without crashing', () => {
+    expect(shallowWrapper.length).toBe(1);
+  });
+  it('renders loading indicator if `isLoading is true`', () => {
+    const wrapper = shallowWrapper;
+    wrapper.setProps({ isLoading: true});
+    expect(wrapper.find('#loading').length).toBe(1);
+  });
+  it('should render the right number of checklist items', () => {
+    // console.log('🔥', shallowWrapper.debug());
```
Comment (seunzone) [inline src/views/Checklist/__tests__/index.js]: this console.log statement should not be in the codebase

**Comment by samuelkubai** [comment]
@clintfidel please define how to make ourselves administrators.

**Comment by clintfidel** [comment]
ok @samuelkubai , will do that

**Comment by clintfidel** [comment]
@samuelkubai, it has been updated now

**Comment by samuelkubai** [comment]
@clintfidel please resolve this conflicts.

**Comment by Nedson202** [comment]
This is being done.

**Comment by samuelkubai** [comment]
@Nedson202 complete?

**Comment by Nedson202** [comment]
Yes it is. @samuelkubai

**Pull request merged**